### ISSUE TITLE
Gmail is not defined when you log in to G Mail.

### ISSUE THREAD

**Issue by SickyBuffalo** [KartikTalwar/gmail.js#383]
When you log out from G Mail and then choose again with wich profile to log in and must type your password and boom Uncaught ReferenceError: Gmail is not defined ! 
![screenshot_1](https://cloud.githubusercontent.com/assets/10416074/24217440/7c2ecb12-0f48-11e7-9a96-d77991a0cd96.jpg)

**Comment by josteink** [comment]
I'm not sure I understand the problem here.  From what I can tell you're either:

* not on `mail.google.com` (so your content-script shouldn't be running anyway)
* or not logged into a Google account (so `mail.google.com` or not, you won't be able to access gmail)

In neither of those cases can `gmail.js` be used. If there's any error here, to me it looks like the error is trying to load `gmail.js` in a situation where it won't work.

Or do I misunderstand something really basic about this situation?

**Comment by SickyBuffalo** [comment]
I was wondering why fires error here, but do not fire error for example when you choose between which account to choose or aby other url ?
![screenshot_1](https://cloud.githubusercontent.com/assets/10416074/24247965/2ba61c48-0fd6-11e7-8940-35b68b5bab28.jpg)

**Comment by josteink** [comment]
Can you show me your manifest.json? I'm pretty sure I can use that to help identify and highlight the cause of this problem.

**Comment by SickyBuffalo** [comment]
Oooh yes, 
"content_scripts": [
    {
      "matches": ["https://mail.google.com/*", "http://*/*", "https://*/*"],
      "js": ["scripts/jquery-2.1.4.min.js","scripts/content.js"]
    }
  ],
Here was the problem !
Thank you very much, i really appreciate your fast answers and help !!!

**Comment by josteink** [comment]
Yes. Your `matches`-clause will result in your content-script getting run on pretty much *all* webpages. If your content-script injects `gmail.js` anywhere except `mail.google.com`, it will fail. There's nothing for it to do there!

Basically your fix needs to be to target your `matches`-clause to be like this: https://github.com/josteink/gmailjs-node-boilerplate/blob/master/manifest.json#L10-L18

If your extension needs to run stuff on other pages than Gmail, you need to isolate that logic in *other* content-scripts (which can have a wider domain mask that Gmail alone).

So basically, I'm going to call this "not a gmail.js" bug and close it, but I wish you luck in getting your project up and running ;)
